http-generator-javalin/src/main/java/io/avaje/http/generator/javalin/ControllerWriter.java

```diff
@@ -87,7 +87,13 @@ private void writeClassStart() {
       writer.append("  private final Validator validator;").eol();
     }
 
-    for (final UType type : jsonTypes.values()) {
+    for (UType type : jsonTypes.values()) {
+      // Support for CompletableFuture's.
+      if (type.isGeneric() && type.mainType().equals("java.util.concurrent.CompletableFuture")) {
```

Comment:
you need to add a check here so that the raw type of the future doesn't get written twice.

Reply:
I guess either modify the `jsontypes` map, or create a set that tracks what future types have been written. Other non- types do not get duplicates.

Reply:
I'm thinking of doing something like this. 
```
      for (UType type : jsonTypes.values()) {
        // Support for CompletableFuture's.
        if ("java.util.concurrent.CompletableFuture".equals(type.mainType())) {
          type = type.paramRaw();
          if (jsonTypes.containsKey(type.full()) || !futureSet.add(type.full())) {
            continue;
          }
        }
```

Reply:
You'll probably need to take out the` throw IllegalStateException` from the JsonbUtil and put it into the first for loop in the ControllerWriter.

Reply:
Now sure how I didn't notice the `jsonTypes` map and that it could be used to fix the issue I came across.

I've moved the `CompletableFuture` generic validation over to the `ControllerMethodWriter` for Javalin as it should then provide better context (controller + method) in regards to _what_ is wrongly defined.

Async tests in `test-javalin-json` have also been added.